projects/cm_mcu/LocalTasks.c

```diff
@@ -1155,7 +1231,9 @@ void init_registers_ff()
   apollo_i2c_ctl_reg_w(3, 0x21, 1, 0x02, 1, 0x00); //  00000000 [P07..P00]
   apollo_i2c_ctl_reg_w(3, 0x21, 1, 0x03, 1, 0x01); //  00000001 [P17..P10]
 
-  xSemaphoreGive(ffldaq_f1_args.xSem); // if we have a semaphore, give it
+  setFFmask(present_FFLDAQ_F1, present_FFL12_F1, present_FFLDAQ_F2, present_FFL12_F2);
```

Comment:
I am not sure if it makes sense to have these commands (which read the `present` pins for the fireflies) in this function makes sense for a function that is called `init_ff_regs`. It seems like a 'side effect' (not in the strict sense, but in the sense that if you read the function name `init_ff_regs` you do not expect it to check the `present` pins.

Reply:
also, the arguments to this function are all 8 bits/1 byte each; maybe just concat them to one single mask rather than having 4 masks?

Reply:
this would also simplify the logic within `setFFmask`

Reply:
I can try to make this part cleaner by creating another function called `readFFpresent()` and will make sure to combine all four 8-bit masks to one argument to the `setFFmask()`